### PR TITLE
Add torch.no_grad() to prevent the GPU memory from keep increasing 

### PR DESCRIPTION
**Pull request by hammer-wang**
Hi, 

I found that when doing manual gradient update, PyTorch will keep growing the computation graph. If dealing with extremely long data stream, the model will consume too much GPU memory. 

I added one line at line 83 to stop gradient computation when manual update the model weights. I tested the code and confirmed that the GPU memory no longer grows as the model ingests the data stream.

### CODE REVIEW

**Comment by alison-carrera** [comment]
@hammer-wang Thanks! I'm going to accept this soon.

**Comment by alison-carrera** [comment]
Hi @hammer-wang did your model converged? Using the torch.no_grad() technically would turn off the backward function and the model wouldn't be able to converge. I'll see it more carefully. Thanks!

**Comment by alison-carrera** [comment]
I don't have a machine with GPU to test a big input data stream. I'll need your help to fix this issue. Thanks!

**Comment by hammer-wang** [comment]
Hi @alison-carrera, thanks a lot for replying! Yes, my model did converge. The reason why the code snippet works is because loss was calculated outside of the torch.no_grad() scope and it's possible to backward through the computation graph. To verify it, you can run a simple code snippet here, 

```
a = torch.tensor([1.,2.,3.], requires_grad=True)  
b = torch.tensor([5.,6.,7.], requires_grad=True)  
c = torch.sum(a + b)
with torch.no_grad():
    c.backward()
    print(a.grad.data)
```

The printed gradient will be [1,1,1]. If the torch.no_grad() stops the gradient computation, you would expect None as the gradient. 

torch.no_grad() is essential for using the code with GPU, because without stopping gradient, the manual gradient update will keep building the computation the graph, which will eventually lead to GPU memory leakage. 

I hypothesize that you can verify this effect using CPU. If you run the code on a really long stream, you should be able to see that the RAM usage keeps increasing. 

Hope this helps!

**Comment by alison-carrera** [comment]
Thanks for your explanation @hammer-wang! Your modification is in master and a new package on pip was released. Really thanks for your contribution!
